wtxmgr/doc.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2013-2015 Conformal Systems LLC <info@conformal.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+// TODO: This package doco needs to be updated for the new wtxmgr API.
+//
+// Package wtxmgr provides an implementation of a transaction store for a
+// bitcoin wallet.  Its primary purpose is to save transactions with outputs
+// spendable with wallet keys and transactions that are signed by wallet keys in
+// memory, handle spend tracking for newly-inserted transactions, and report the
+// spendable balance from each unspent transaction output. It uses walletdb as
+// the backend for storing the serialized transaction objects in buckets.
+//
+// Transaction outputs which are spendable by wallet keys are called credits
+// (because they credit to a wallet's total spendable balance) and are modeled
+// using the Credit structure.  Transaction inputs which spend
+// previously-inserted credits are called debits (because they debit from the
+// wallet's spendable balance) and are modeled using the Debit structure.
+//
+// Besides just saving transactions, bidirectional spend tracking is also
+// performed on each credit and debit.  Unlike packages such as btcdb, which
```

Comment:
This entire file was lifted from the old implementation and needs to be rewritten.